### DISCUSSION TITLE
when does kadalu will supoort snapshot , clone of gluster volume ?

### DISCUSSION THREAD

**Discussion by linpanru** [kadalu/kadalu#992] [Q&A]
when does kadalu will supoort snapshot , clone of gluster volume ?

**Comment by aravindavk** [comment]
Gluster volume is used as Storage pool when integrated with Kubernetes. One Storage pool (Gluster volume) can provision many PVs. It doesn't make sense to take a full Storage pool Snapshot when a single PV snapshot is requested.

For example, if a Storage Pool contains 256 or more PVs then we hit the Gluster Snapshot limit when all PVs snapshots are taken or half number of PVs take snapshots twice.

Another possibility we are exploring is use of XFS reflink copy feature. Write an orchestrator that adds a barrier to the PV (Pause the I/O) and takes reflink copy of each files in the PV from all bricks.  The design is not finalized, and this is only limited to XFS backends.

**Reply by linpanru** [reply]
> Gluster volume is used as Storage pool when integrated with Kubernetes. One Storage pool (Gluster volume) can provision many PVs. It doesn't make sense to take a full Storage pool Snapshot when a single PV snapshot is requested.
> 
> For example, if a Storage Pool contains 256 or more PVs then we hit the Gluster Snapshot limit when all PVs snapshots are taken or half number of PVs take snapshots twice.
> 
> Another possibility we are exploring is use of XFS reflink copy feature. Write an orchestrator that adds a barrier to the PV (Pause the I/O) and takes reflink copy of each files in the PV from all bricks. The design is not finalized, and this is only limited to XFS backends.

Thanks very much for your explanation
